src/modules/powerrename/lib/Helpers.cpp

```diff
@@ -1,7 +1,49 @@
 #include "pch.h"
 #include "Helpers.h"
+#include <regex>
 #include <ShlGuid.h>
 
+HRESULT GetDatedFileName(_Out_ PWSTR result, UINT cchMax, _In_ PCWSTR source, SYSTEMTIME LocalTime)
+{
+    HRESULT hr = (source && wcslen(source) > 0) ? S_OK : E_INVALIDARG;     
+    if (SUCCEEDED(hr))
+    {
+        std::wregex pattern(L"\\$YYYY");
+        std::wstring res = source;
+        wchar_t replaceTerm[MAX_PATH] = {0};
+        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%d"),LocalTime.wYear);
+        res = regex_replace(std::wstring(res), pattern, replaceTerm);
+
+        pattern = L"\\$MM" ;
+        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%02d"), LocalTime.wMonth);
+        res = regex_replace(std::wstring(res), pattern, replaceTerm);
+
+        pattern = L"\\$DD";
+        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%02d"), LocalTime.wDay);
+        res = regex_replace(std::wstring(res), pattern, replaceTerm);
+
+        pattern = L"\\$hh";
+        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%02d"), LocalTime.wHour);
+        res = regex_replace(std::wstring(res), pattern, replaceTerm);
+
+        pattern = L"\\$mm";
+        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%02d"), LocalTime.wMinute);
+        res = regex_replace(std::wstring(res), pattern, replaceTerm);
+
+        pattern = L"\\$ss";
+        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%02d"), LocalTime.wSecond);
+        res = regex_replace(std::wstring(res), pattern, replaceTerm);
+
+        pattern = L"\\$SSS";
```

Comment:
Is `SSS` common for milliseconds? I would have expect something like `MMM` or `mmm`.

Reply:
SSS is stated by the issue author. I am not sure which one is most common. [Some sources](https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings) also states it as `fff`. Do yo think covering all `MMM`, `SSS`, `FF`, `ff` options could be better?